lib/Dialect/FIRRTL/Ops.cpp

```diff
@@ -74,6 +75,118 @@ static LogicalResult verifyCircuitOp(CircuitOp &circuit) {
     return failure();
   }
 
+  // Store a mapping of defname strings to the first external module
+  // that defines it.
+  llvm::StringMap<FExtModuleOp> defnameMap;
```

Comment:
Thank you for using efficient algorithms, however you can do one better here: instead of indexing with the string itself, you can use an `llvm::DenseMap<Attribute, FExtModuleOp>`.  mlir::Attribute is a uniqued thing a pointer in size, so this will be very efficient.

Reply:
Done. Back to the `DenseMap` you originally suggested. :)